src/app/[lang]/verify-request/verify-token-input.tsx

```diff
@@ -0,0 +1,70 @@
+"use client";
+
+import { useState } from "react";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { useRouter } from "next/navigation";
+import type { Dictionary } from "@/dictionaries/dictionary";
+
+interface VerifyTokenInputProps {
+    dictionary: Dictionary;
+}
+
+export function VerifyTokenInput({ dictionary }: VerifyTokenInputProps) {
+    const [token, setToken] = useState("");
+    const [isLoading, setIsLoading] = useState(false);
+    const router = useRouter();
+
+    const handleVerification = async () => {
+        try {
+            setIsLoading(true);
+            const emailParam = new URLSearchParams(window.location.search).get("email");
+            const callbackUrl = new URLSearchParams(window.location.search).get("callbackUrl");
+            const decodedEmail = emailParam ? decodeURIComponent(emailParam) : "";
+
+            // Construct the verification URL similar to the email link
```

Comment:
There are some more comments like this ;) in the code. Do we want to keep them?

Reply:
these comments are reasoning and not solely explanations
If you change the way the Url is constructed -> doesn't work
if you change the way we forward to url -> doesn't work

we can remove them but i would advise otherwise